lib/livebook/notebook.ex

```diff
@@ -298,9 +335,17 @@ defmodule Livebook.Notebook do
   """
   @spec parent_cells_with_section(t(), Cell.id()) :: list({Cell.t(), Section.t()})
   def parent_cells_with_section(notebook, cell_id) do
+    parent_cell_ids =
+      notebook
+      |> cell_dependency_graph()
+      |> Graph.find_path(cell_id, nil)
+      |> MapSet.new()
+      |> MapSet.delete(cell_id)
+      |> MapSet.delete(nil)
```

Comment:
Why would we have nil here? 🤔

Reply:
`Graph.find_path` returns a list of nodes including both the starting end ending point, and in this case `nil` is our ending point.